### PR TITLE
register: default to admin `search_fields`, fallback to 'name'/'title'

### PR DESCRIPTION
**Pull request by blueyed**
Followup of https://github.com/yourlabs/django-autocomplete-light/pull/301 against master.

Fixes https://github.com/yourlabs/django-autocomplete-light/issues/239.

### CODE REVIEW

**Comment by blueyed** [comment]
This should be called after all other apps have registered any admin classes for their models, so that `search_fields` is actually available.

That could be achieved by having it after django.contrib.admin in INSTALLED_APPS, but the documentation says that it's required to be in front.

Could search_field become a property, that looks up the default via the admin's registry on demand/later?

**Comment by jpic** [comment]
> This should be called after all other apps have registered any admin classes for their models, so that search_fields is actually available.

Good point ^^

> That could be achieved by having it after django.contrib.admin in INSTALLED_APPS, but the documentation says that it's required to be in front.

Well the admin needs forms, and forms need autocomplete_light_registries to be loaded for any autocomplete action to happen.

> Could search_field become a property, that looks up the default via the admin's registry on demand/later?

This could work, after all, search_fields powers choices_for_request() which is used by autocomplete_html(), which is called by AutocompleteView. In theory, there is no need for search_fields to be configured upfront. 

You'd like to try making AutocompleteModel.search_fields an `@property` method ? Makes sense to me.

**Comment by blueyed** [comment]
> You'd like to try making AutocompleteModel.search_fields an @property method ? Makes sense to me.

Yes.

The drawback I see is that it would not result in errors (`UnconfiguredSearchFieldsException`) during startup, but during runtime.

Is there another possibility, e.g. a hook / signal after all apps have been loaded?

**Comment by jpic** [comment]
> Is there another possibility, e.g. a hook / signal after all apps have been loaded?

Actually, there is a new feature which I haven't yet looked at in details: https://docs.djangoproject.com/en/1.7/ref/checks/

Would that work ?

**Comment by blueyed** [comment]
That would work probably.

**Comment by jpic** [comment]
Sounds like the next release is all yours!
Rock'on

File: autocomplete_light/registry.py [hunk 1]
```diff
@@ -173,12 +176,20 @@ def _register_model_autocomplete(self, model, autocomplete=None,
 
         if base.search_fields is None and 'search_fields' not in kwargs:
             try:
-                model._meta.get_field('name')
-            except:
-                raise Exception('Add search_fields kwargs to .register(%s)'
-                                % model.__name__)
-            else:
-                kwargs['search_fields'] = ['name']
+                search_fields = admin.site._registry[model].search_fields
```
Comment (jpic) [inline autocomplete_light/registry.py]: We might be using the wrong site though, we need the admin_site from there: https://github.com/django/django/blob/stable/1.8.x/django/contrib/admin/widgets.py#L241

**Pull request closed without merging**